### PR TITLE
Mean Squared Error (MSE) Metric Document

### PR DESCRIPTION
**Pull request by sdhossain**
### Linked issue(s)

<s>Requires [#549](https://github.com/kolenaIO/kolena/pull/549) to be merged before. (Will need to rebase/add links)</s>

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: docs/metrics/mean-squared-error.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Mean Squared Error
+
+Mean Squared Error (MSE) is a widely used metric for evaluating the performance of regression models,
+popular for its ability to accentuate larger errors.
+
+MSE measures the average squared difference between the predicted values and the actual values,
+emphasizing larger errors more significantly than smaller ones. It represents the mean of the squared
+discrepancies between predicted and actual values across a dataset, penalizing larger errors more heavily.
```
Comment (y27choi) [inline docs/metrics/mean-squared-error.md]: Let's include how to interpret this metric.  What does it mean to have a high MSE?  (I forgot to comment on this in the MAE doc)

File: docs/metrics/mean-squared-error.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+$$
+\begin{align}
+\text{MSE} &= \frac{(25 - 27)^2 + (35 - 30)^2}{2} \\
+&= 14.5
+\end{align}
+$$
+</div>
+
+Age Estimation::
```
Comment (y27choi) [inline docs/metrics/mean-squared-error.md]: double colon (also on MAE doc)

File: docs/metrics/mean-squared-error.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+## Limitations and Biases
+
+While Mean Squared Error (MSE) offers a clear metric for evaluating regression model accuracy by heavily penalizing
+larger errors, this same feature can also be seen as a limitation. MSE can disproportionately represent the effect of
+outliers or extreme errors on the overall model performance, potentially leading to a skewed perception of a model's
+predictive ability.
+
+In contexts where outliers are significant or the distribution of errors is important, relying solely on MSE may not
+provide a fully accurate evaluation. It is crucial to supplement MSE with other metrics, such as Mean Absolute Error
```
Comment (y27choi) [inline docs/metrics/mean-squared-error.md]: Let's add a link to the MAE doc here.

File: docs/metrics/index.md [hunk 4]
```diff
@@ -236,6 +236,14 @@ biases, and its intended uses.
     the mean of the absolute differences between predicted and actual values across a dataset, treating each
     discrepancy equally.
 
+- [Mean Squared Error](mean-squared-error.md)
+
+    ---
+
+    MSE measures the average squared difference between the predicted values and the actual values,
+    emphasizing larger errors more significantly than smaller ones. It is mainly different from MAE in that
```
Comment (y27choi) [inline docs/metrics/index.md]: Let's add a link to MAE doc where MAE is mentioned.

File: docs/metrics/mean-squared-error.md (hunk 1 above)
Comment (sdhossain) [reply docs/metrics/mean-squared-error.md]: Do we want to add a sentence in the `Examples` section on this as well? (Say how an MAE of 4 means that the predictor is on average 4 years off the actual age, etc.) - I've appended the basic understanding of it in the first section.

**Review by y27choi — APPROVED** [review]

File: docs/metrics/mean-squared-error.md (hunk 1 above)
Comment (y27choi) [reply docs/metrics/mean-squared-error.md]: That could be nice.

**Comment by y27choi** [comment]
As it was discussed in [this PR thread](https://github.com/kolenaIO/kolena/pull/561#discussion_r1547923830), let's move this doc to "Common Metrics".

**Comment by sdhossain** [comment]
> As it was discussed in [this PR thread](https://github.com/kolenaIO/kolena/pull/561#discussion_r1547923830), let's move this doc to "Common Metrics".

would this be where "Accuracy", etc. and others are currently located?

**Comment by y27choi** [comment]
> > As it was discussed in [this PR thread](https://github.com/kolenaIO/kolena/pull/561#discussion_r1547923830), let's move this doc to "Common Metrics".
> 
> would this be where "Accuracy", etc. and others are currently located?

Correct.  It will be added to this section of the doc.
image

File: docs/metrics/index.md [hunk 5]
```diff
@@ -103,6 +103,22 @@ biases, and its intended uses.
     are the building blocks of many other metrics, including [accuracy](accuracy.md), [precision](precision.md),
     and [recall](recall.md).
 
+- [Mean Absolute Error](mean-absolute-error.md)
```
Comment (y27choi) [inline docs/metrics/index.md:106]: I believe this section is alphabetically organized.

Comment (sdhossain) [reply docs/metrics/index.md:106]: Oh was not aware - I put some fixes in this PR #566

**Pull request merged**